enhance_crime_validation.py

```
@@ -0,0 +1,494 @@
+                    tables_to_check = [
+                        'crime_density_grid',
+                        'crime_scores', 
+                        'enhanced_scored_properties',
+                        'scoring_results'
+                    ]
+                    
+                    found_data = False
+                    
+                    for table in tables_to_check:
+                        try:
+                            cursor.execute(f"SELECT name FROM sqlite_master WHERE type='table' AND name='{table}'")
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)